pkg/monitors/jenkins/nodemetrics.go

```diff
@@ -0,0 +1,99 @@
+package jenkins
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	jc "github.com/signalfx/signalfx-agent/pkg/monitors/jenkins/client"
+	"github.com/signalfx/signalfx-agent/pkg/monitors/types"
+)
+
+// map key:metric_name value:json_key
+var nodeMetricsMap = map[string]string{
+	"jenkins.node.vm.memory.total.used":    "vm.memory.total.used",
+	"jenkins.node.vm.memory.heap.usage":    "vm.memory.non-heap.used",
+	"jenkins.node.vm.memory.non-heap.used": "vm.memory.non-heap.used",
+	"jenkins.node.queue.size":              "jenkins.queue.size.value",
+	"jenkins.node.health-check.score":      "jenkins.health-check.score",
+	"jenkins.node.executor.count":          "jenkins.executor.count.value",
+	"jenkins.node.executor.in-use":         "jenkins.executor.in-use.value",
+}
+
+type NodeMetrics struct {
+	Version string                            `json:"version"`
+	Gauges  map[string]map[string]interface{} `json:"gauges"`
+}
+
+func nodeMetrics(jkClient jc.JenkinsClient, conf *Config) ([]*datapoint.Datapoint, error) {
+	var node NodeMetrics
+	err := jkClient.FetchJSON(fmt.Sprintf(metricsEndpoint, conf.MetricsKey), &node)
+	if err != nil {
+		return nil, err
+	}
+	dps := getNodeMetricsDataPoints(&node)
+	return dps, nil
+}
+
+func getNodeMetricsDataPoints(node *NodeMetrics) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+	for _, metric := range groupMetricsMap[groupNode] {
+		if jsonKey, ok := nodeMetricsMap[metric]; ok {
+			if val, ok := node.Gauges[jsonKey]; ok {
+				v, err := datapoint.CastMetricValue(val["value"])
+				if err != nil {
+					// Could be a string
+					logger.Warnf("Skipping metric: failed to cast value for key %s and value %v : %v", metric, val, err)
+					continue
+				}
+				dps = append(dps, datapoint.New(metric, nil, v, metricSet[metric].Type, time.Time{}))
+			} else {
+				logger.Warnf("Failed to find json key for metric: %s", metric)
+			}
+		}
+	}
+	return dps
+}
+
+func liveness(jkClient jc.JenkinsClient, conf *Config) []*datapoint.Datapoint {
+	var val int64
+	result, err := jkClient.FetchText(fmt.Sprintf(pingEndPoint, conf.MetricsKey))
+	if err != nil {
+		logger.WithError(err).Warnf("Failed to ping the jenkins server, sending status 0")
+		val = 0
+	}
+	if strings.TrimSpace(result) == "pong" {
+		val = 1
+	}
+	var dps []*datapoint.Datapoint
+	dps = append(dps, datapoint.New(jenkinsNodeOnlineStatus, nil, datapoint.NewIntValue(val), metricSet[jenkinsNodeOnlineStatus].Type, time.Time{}))
+
+	return dps
+}
+
+func getUniqueDimension(jkClient jc.JenkinsClient, conf *Config) (*types.Dimension, error) {
+	nodeInfo, err := jkClient.GoJenkins.Info()
+	if err != nil {
+		return nil, err
+	}
+
+	dim := &types.Dimension{
+		Name:              "jenkins_cluster",
+		Value:             conf.JenkinsCluster,
+		Tags:              map[string]bool{},
```

Comment:
See if there are any other useful values that could be sent as properties.

Reply:
e.g. jenkins version.

Reply:
@keitwb I couldn't find the jenkins version through api, I will keep digging https://github.com/signalfx/signalfx-agent/pull/1191#discussion_r377748412

Reply:
@keitwb found the version, it's in the header ("X-Jenkins"), I will add it as properties

Reply:
done